File: solutions/csharp/series/1/Series.cs
public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength <= 0 || sliceLength > numbers.Length)
        {
            throw new ArgumentException();
        }
        
        var slices = new List<string>();
        var digits = numbers.ToCharArray();

        for (var i = 0; i < digits.Length - sliceLength + 1; i++)
        {
            slices.Add(new string(digits[i..(i + sliceLength)]));
        }

        return [.. slices];
    }
}

File: solutions/elixir/largest-series-product/2/lib/series.ex
defmodule Series do
  @doc """
  Finds the largest product of a given number of consecutive numbers in a given string of numbers.
  """
  @spec largest_product(String.t(), non_neg_integer) :: non_neg_integer
  def largest_product(number_string, size) do
    number_string
    |> String.graphemes()
    |> Enum.map(&String.to_integer/1)
    |> largest_product2(size)
  end

  defp largest_product2(digits, size) when length(digits) < size, do: raise ArgumentError
  defp largest_product2(_, size) when size <= 0, do: raise ArgumentError

  defp largest_product2(digits, size) do
    spans = digits
    |> Enum.chunk_every(size, 1, :discard)

    spans
    |> Enum.map(&product_of_digits/1) |> Enum.max()
  end

  defp product_of_digits(digits), do: Enum.reduce(digits, 1, fn(x, acc) -> x * acc end)
end
